version: "3"

services:
  mysql:
    container_name: mysql
    hostname: mysql
    image: psyoblade/data-engineer-mysql:1.1
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: testdb
      MYSQL_USER: sqoop
      MYSQL_PASSWORD: sqoop
    ports:
      - '3306:3306'
    networks:
      - default
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      interval: 3s
      timeout: 1s
      retries: 3
    volumes:
      - ./mysql/etc:/etc/mysql/conf.d
  sqoop:
    container_name: sqoop
    hostname: sqoop
    image: psyoblade/data-engineer-sqoop:2.0
    tty: true
    environment:
      LANG: ko_KR.UTF-8
    ports:
      - '8042:8042'
      - '8088:8088'
      - '10020:10020'
      - '19888:19888'
      - '60070:50070'
      - '60075:50075'
    networks:
      - default
    depends_on:
      - mysql
    volumes:
      - ./sqoop/jars:/jdbc
      - ./tmp/target:/tmp/target
      - ./mysql/mysqldump:/usr/local/sqoop/bin/mysqldump

networks:
  default:
    name: data_engineer_intermediate_day2_network
